goodreads-predictor/src/goodreads_predictor/pipelines/feature_engineering/nodes.py

```diff
@@ -0,0 +1,233 @@
+    count_data[bin_col_name] = output_array
+
+    return count_data.drop(columns = 'count'), mean_count, mean_count_no_1
+
+def calculate_author_fame_levels(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Calculate the fame levels of authors based on their total review count.
+
+    Args:
+        data (pd.DataFrame): The input DataFrame containing the data.
+
+    Returns:
+        pd.DataFrame: A DataFrame containing the fame levels of authors.
+                      The DataFrame has a single column named 'AuthorFameLevel'.
+
+    Notes:
+        The fame levels are mapped as follows:
+        - 1: "1 - Very Low Exposure"
+        - 2: "2 - Not Well-Known"
+        - 3: "3 - Well-Known"
+        - 4: "4 - Popular"
+        - 5: "5 - Semi-Famous"
+        - 6: "6 - Famous"
+    """
+    author_fame_levels = ...
+    # Use percentiles of the total review count to get a sense of how well-known an author is
+    author_fame_levels = (
+        data
+        # Get total review count by author
+        .groupby("most_common_author")
+        .agg(TotalReviews = ("ratings_count", "sum"))
+        .sort_values("TotalReviews", ascending=False)
+
+        # Get quantiles of the total review count (list comprehension to repeat the quantiles for each author)
+        .assign(QuantileReviews = lambda x: [x['TotalReviews'].quantile([0, 0.15, 0.25, 0.5, 0.75, 0.95]).values.tolist() for _ in range(len(x))])
+
+        # Explode the quantiles to get one row per quantile then sort by quantile
+        .explode("QuantileReviews")
+        .sort_values("QuantileReviews", ascending=True)
+
+        # Create a quantile index for each author
+        .assign(QuantileRank = lambda x: x.groupby("most_common_author").cumcount().add(1))
+
+        # Filter to where the total review count is greater than the quantile review count or we are at the lowest quantile possible
+        .assign(ReviewsDiff = lambda x: x['TotalReviews'] - x['QuantileReviews'])
+        .assign(MaxReviewsDiff = lambda x: x.groupby("most_common_author")["ReviewsDiff"].transform("max"))
+        .assign(AbsReviewsDiff = lambda x: abs(x["ReviewsDiff"]))
+        .query("ReviewsDiff > 0 or MaxReviewsDiff <= 0")
+
+        # Get the first row for each author (the lowest quantile where the total review count is greater)
+        .sort_values("AbsReviewsDiff", ascending=True)
+        .groupby("most_common_author")
+        .first()
+
+        # Map the quantile rank to a string ("Unknown", "Not Well-Known", "Well-Known", "Very Well-Known", "Famous")
+        .assign(AuthorFameLevel = lambda x: x["QuantileRank"].map({1: "1 - Very Low Exposure", 2: "2 - Not Well-Known", 3: "3 - Well-Known",
+                                                                   4: "4 - Popular", 5:"5 - Semi-Famous", 6: "6 - Famous"}))
+        .sort_values("TotalReviews", ascending=False)
+        [['AuthorFameLevel']]
+    )
+
+    return author_fame_levels
```

Comment:
Nice one! Having this is awesome as it is clearly a must to a priori define what is a good book